tests/platforms/android/apk/test_verify_tools.py

```diff
@@ -0,0 +1,183 @@
+import subprocess
+from io import BytesIO
+from stat import S_IMODE
+from unittest import mock
+from zipfile import ZipFile
+
+import pytest
+from requests import exceptions as requests_exceptions
+
+from briefcase.exceptions import BriefcaseCommandError, NetworkFailure
+from briefcase.platforms.android.apk import ApkBuildCommand
+
+
+def create_sentinel_zipfile():
+    out = BytesIO()
+    with ZipFile(out, "w") as zipfile:
+        zipfile.writestr("tools/bin/sdkmanager", "")
+    return out.getvalue()
+
+
+@pytest.fixture
+def build_command(tmp_path, first_app_config):
+    class TestableApkBuildCommand(ApkBuildCommand):
+        dot_briefcase = tmp_path / ".briefcase" / "tools"
+
+    command = TestableApkBuildCommand(
+        base_path=tmp_path / "base_path", apps={"first": first_app_config},
+    )
+    command.tmp_path = tmp_path
+    command.host_os = "Linux"
+    command.os = mock.MagicMock()
+    command.os.environ = {}
+    command.sys = mock.MagicMock()
+    command.sys.version_info.major = 3
+    command.sys.version_info.minor = 7
+    command.requests = mock.MagicMock()
+    command.subprocess = mock.MagicMock()
+    return command
+
+
+def test_sdk_url(build_command):
+    build_command.host_os = "Linux"
```

Comment:
As above - it would be preferable to set this to a known non-existent value (e.g., "PerfectOS") to check that the URL construction is correct, but *completely* dependent on host_os.

Reply:
OK!